streams/src/test/java/org/apache/kafka/streams/state/internals/CachingWindowStoreTest.java

```diff
@@ -83,6 +96,104 @@ public void closeStore() {
+
+        builder.addStateStore(storeBuilder);
+
+        builder.stream(topic, Consumed.with(Serdes.String(), Serdes.String()))
+                .transform(() -> new Transformer<String, String, KeyValue<String, String>>() {
+
+                    private WindowStore<String, String> store;
+                    private int numRecordsProcessed;
+                    @Override
+                    public void init(final ProcessorContext processorContext) {
+                        this.store = (WindowStore<String, String>) processorContext.getStateStore("store-name");
+                        int count = 0;
+
+                        final KeyValueIterator<Windowed<String>, String> all = store.all();
+                        while (all.hasNext()) {
+                            count++;
+                            all.next();
+                        }
+
+                        assertThat(count, equalTo(0));
+                    }
+
+                    @Override
+                    public KeyValue<String, String> transform(final String key, final String value) {
+                        int count = 0;
+
+                        final KeyValueIterator<Windowed<String>, String> all = store.all();
+                        while (all.hasNext()) {
+                            count++;
+                            all.next();
+                        }
+                        assertThat(count, equalTo(numRecordsProcessed));
+
+                        store.put(value, value);
+
+                        numRecordsProcessed++;
+
+                        return new KeyValue<>(key, value);
+                    }
+
+                    @Override
+                    public void close() {
+
+                    }
+                }, "store-name");
+
+        final String bootstrapServers = "localhost:9092";
+        final Properties streamsConfiguration = new Properties();
+        streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "test-app");
+        // Where to find Kafka broker(s).
+        streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        // Specify default (de)serializers for record keys and for record values.
+        streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+        streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+        streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getPath());
+        // Records should be flushed every 10 seconds. This is less than the default
+        // in order to keep this example interactive.
+        streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 10 * 1000);
+
+        final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), streamsConfiguration);
```

Comment:
We should initialize `TopologyTestDriver` with a fixed mock-timestamp and use this below instead of calling `System. currentTimeMillis()`

Reply:
ack